x-pack/plugins/monitoring/server/rules/cpu_usage_rule.test.ts

```diff
@@ -0,0 +1,334 @@
+import { RULE_CPU_USAGE } from '../../common/constants';
+import { fetchCpuUsageNodeStats } from '../lib/alerts/fetch_cpu_usage_node_stats';
+import { fetchClusters } from '../lib/alerts/fetch_clusters';
+import { alertsMock } from '@kbn/alerting-plugin/server/mocks';
+import { ALERT_REASON } from '@kbn/rule-data-utils';
+
+const RealDate = Date;
+
+jest.mock('../lib/alerts/fetch_cpu_usage_node_stats', () => ({
+  fetchCpuUsageNodeStats: jest.fn(),
+}));
+jest.mock('../lib/alerts/fetch_clusters', () => ({
+  fetchClusters: jest.fn(),
+}));
+jest.mock('../static_globals', () => ({
+  Globals: {
+    app: {
+      getLogger: () => ({ debug: jest.fn() }),
+      url: 'http://localhost:5601',
+      config: {
+        ui: {
+          ccs: { enabled: true },
+          container: { elasticsearch: { enabled: false } },
+        },
+      },
+    },
+  },
+}));
+
+describe('CpuUsageRule', () => {
+  it('should have defaults', () => {
+    const rule = new CpuUsageRule();
+    expect(rule.ruleOptions.id).toBe(RULE_CPU_USAGE);
+    expect(rule.ruleOptions.name).toBe('CPU Usage');
+    expect(rule.ruleOptions.throttle).toBe('1d');
+    expect(rule.ruleOptions.defaultParams).toStrictEqual({ threshold: 85, duration: '5m' });
+    expect(rule.ruleOptions.actionVariables).toStrictEqual([
+      { name: 'node', description: 'The node reporting high cpu usage.' },
+      {
+        name: 'internalShortMessage',
+        description: 'The short internal message generated by Elastic.',
+      },
+      {
+        name: 'internalFullMessage',
+        description: 'The full internal message generated by Elastic.',
+      },
+      { name: 'state', description: 'The current state of the alert.' },
+      { name: 'clusterName', description: 'The cluster to which the node(s) belongs.' },
+      { name: 'action', description: 'The recommended action for this alert.' },
+      {
+        name: 'actionPlain',
+        description: 'The recommended action for this alert, without any markdown.',
+      },
+    ]);
+  });
+
+  describe('execute', () => {
+    function FakeDate() {}
+    FakeDate.prototype.valueOf = () => 1;
+
+    const clusterUuid = 'abc123';
+    const clusterName = 'testCluster';
+    const nodeId = 'myNodeId';
+    const nodeName = 'myNodeName';
+    const cpuUsage = 91;
+    const stat = {
+      clusterUuid,
+      nodeId,
+      nodeName,
+      cpuUsage,
+    };
+
+    const services = alertsMock.createRuleExecutorServices();
+    const executorOptions = {
+      services,
+      state: {},
+    };
+
+    beforeEach(() => {
+      // @ts-ignore
+      Date = FakeDate;
+      (fetchCpuUsageNodeStats as jest.Mock).mockImplementation(() => {
+        return [stat];
+      });
+      (fetchClusters as jest.Mock).mockImplementation(() => {
+        return [{ clusterUuid, clusterName }];
+      });
+    });
+
+    afterEach(() => {
+      Date = RealDate;
+      jest.resetAllMocks();
+    });
+
+    it('should fire actions', async () => {
```

Comment:
Nit: should this be `an action` (not `actions`) because I think here we have a single action - I see that we are checking for `toHaveBeenCalledTimes(1)`

Reply:
Updated in https://github.com/elastic/kibana/pull/174169/commits/d1904fec084013ffd28f20a7f47be655e50d18a2